js/ui/map.js

```diff
@@ -399,51 +428,56 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
-     * location.
+     * Returns a [`Point`](#Point) representing pixel coordinates, relative to the map's `container`,
+     * corresponding to the specified geographical location.
      *
-     * @param {LngLat} lnglat
-     * @returns {Object} `x` and `y` coordinates
+     * @param {LngLatLike} lnglat The geographical location to project.
+     * @returns {Point} The [`Point`](#Point) corresponding to `lnglat`, relative to the map's `container`.
      */
     project: function(lnglat) {
         return this.transform.locationPoint(LngLat.convert(lnglat));
     },
 
     /**
-     * Get geographical coordinates, given pixel coordinates.
+     * Returns a [`LngLat`](#LngLat) representing geographical coordinates corresponding
+     * to the specified pixel coordinates.
      *
-     * @param {Array<number>} point [x, y] pixel coordinates
-     * @returns {LngLat}
+     * @param {PointLike} point The pixel coordinates to unproject.
+     * @returns {LngLat} The [`LngLat`](#LngLat) corresponding to `point`.
      */
     unproject: function(point) {
         return this.transform.pointLocation(Point.convert(point));
     },
 
     /**
-     * Query rendered features at a point or within a rectangle.
+     * Returns an array of [GeoJSON](http://geojson.org/)
+     * [Feature objects](http://geojson.org/geojson-spec.html#feature-objects)
+     * representing visible features that satisfy the query parameters.
      *
-     * @param {Point|Array<number>|Array<Point>|Array<Array<number>>} [pointOrBox] - The geometry of a query region:
-     * either [x, y] pixel coordinates of a point, or [[x1, y1], [x2, y2]] pixel coordinates of opposite corners of
-     * a bounding rectangle. Omitting this parameter (i.e. calling `queryRenderedFeatures` with zero arguments,
-     * or with a single `params` argument), is equivalent to passing a bounding rectangle encompassing the entire
-     * viewport.
-     * @param {Object} [params]
-     * @param {Array<string>} [params.layers] Only query features from layers with these layer IDs.
-     * @param {Array} [params.filter] A [filter](https://www.mapbox.com/mapbox-gl-style-spec/#types-filter).
+     * @param {PointLike|Array<PointLike>} [region] - The geometry of the query region:
+     * either a single point or southwest and northeast points describing a bounding box.
+     * Omitting this parameter (i.e. calling [`Map#queryRenderedFeatures`](#Map#queryRenderedFeatures) with zero arguments,
+     * or with only a `parameters` argument) is equivalent to passing a bounding box encompassing the entire
+     * map viewport.
+     * @param {Object} [parameters]
+     * @param {Array<string>} [parameters.layers] An array of style layer IDs for the query to inspect.
+     *   Only features within these layers will be returned. If this parameter is undefined, all layers will be checked.
+     * @param {Array<string>} [parameters.filter] A [filter](https://www.mapbox.com/mapbox-gl-style-spec/#types-filter)
```

Comment:
Let's leave this type parameter as `Array`. `Array<string>` doesn't accurately capture filters like `['any', ['==', '@penny', true], ['==', '@linus', true]]`